### PR TITLE
Bug 6223 - Thematic Maps/Saved Maps Enhancement 

### PR DESCRIPTION
**Pull request by wmbeers**
# Description
Separates saved maps functionality into new Saved Maps widget, and adds "Thematic Maps" interface.

### CODE REVIEW

File: viewer/index.html [hunk 1]
```diff
@@ -20,8 +20,7 @@
     <link rel="stylesheet" type="text/css" href="css/jquery-ui.min.css" />
     <script src="js/arrayFind.js"></script>
     <script src="js/string.polyfill.js"></script>
-    <script src="js/knockout-latest.js"></script>
-
+    <script src="https://knockoutjs.com/downloads/knockout-3.4.2.debug.js"></script>
```
Comment (wmbeers) [inline viewer/index.html]: This change  is result of merge conflict resolved with Thematic Maps branch, but should have been estMap. Need to revert.

File: viewer/js/gis/dijit/AoiEditor.js [hunk 2]
```diff
@@ -352,9 +352,11 @@ function (declare, _WidgetBase, _TemplatedMixin, _WidgetsInTemplateMixin,
         },
 
         saveAndShowFeatureList: function () {
+            //TODO validate aoi name and type have been selected
```
Comment (wmbeers) [inline viewer/js/gis/dijit/AoiEditor.js]: All changes in this file should be reverted.

File: viewer/js/gis/dijit/LayerControl/plugins/legendUtil.js [hunk 3]
```diff
@@ -233,10 +233,16 @@ define([
         // create legends for vector layers (feature, etc)
         _createVectorLegend: function (infos, layer, expandNode) {
             //create legend table
-            var table = domConst.create('table');
+            var table = domConst.create('table'),
```
Comment (wmbeers) [inline viewer/js/gis/dijit/LayerControl/plugins/legendUtil.js:236]: The purpose of this change is to handle behavior of certain feature layers in the layer control's legend (AKA TOC). There are certain layers, including some used for WATERSS, which is how we came to discover this here, that show up with multiple, repeated labels.

File: viewer/js/knockout-latest.js [hunk 4]
```diff
@@ -4,121 +4,3540 @@
  * License: MIT (http://www.opensource.org/licenses/mit-license.php)
  */
```
Comment (wmbeers) [inline viewer/js/knockout-latest.js]: None of these changes belong here.

**Review by wmbeers — COMMENTED** [review]
Reviewed all files and looks good after partial reverts.

**Review by mikekonikoff — APPROVED** [review]
Changes look good and change request has been approved.

**Pull request merged**